src/main/java/sk/hfa/images/services/ImageService.java

```diff
@@ -1,172 +1,74 @@
-        } else if (imageType.equals(ImageType.GALLERY_FLOOR_PLANS_IMAGES.getImageType())) {
-            return ImageType.GALLERY_FLOOR_PLANS_IMAGES;
-        } else if (imageType.equals(ImageType.GALLERY_IMAGES.getImageType())) {
-            return ImageType.GALLERY_IMAGES;
-        } else if (imageType.equals(ImageType.BLOG_ARTICLE_TITLE_IMAGE.getImageType())) {
-            return ImageType.BLOG_ARTICLE_TITLE_IMAGE;
-        } else {
-            throw new IllegalArgumentException("Invalid image type");
-        }
+    public Image save(MultipartFile multipartFile) {
+        Image image = imageRepository.save(Image.build(multipartFile));
+        ImageUtils.uploadImageToFileSystem(multipartFile, image);
+        return image;
     }
 
     @Override
-    public void deleteProjectImages(Long projectId) {
-        Project project = projectService.findById(projectId);
-
-        if (project instanceof CommonProject) {
-            deleteImagesByPaths(((CommonProject) project).getFloorPlanImagePaths());
-            ((CommonProject) project).setFloorPlanImagePaths(null);
+    public List<Image> save(List<MultipartFile> multipartFiles) {
+        List<Image> images = new ArrayList<>();
+        for (MultipartFile multipartFile : multipartFiles) {
+            Image image = save(multipartFile);
+            images.add(image);
         }
-        deleteImagesByPaths(project.getImagePaths());
-        project.setImagePaths(null);
-        deleteImagesByPaths(Collections.singletonList(project.getTitleImage()));
-        project.setTitleImage(null);
-        projectService.save(project);
+        return images;
     }
 
     @Override
-    public void deleteBlogArticleImage(Long blogArticleId) {
-        BlogArticle blogArticle = blogService.findById(blogArticleId);
-        if (Objects.isNull(blogArticle.getTitleImage()))
-            return;
-        deleteImagesByPaths(Collections.singletonList(blogArticle.getTitleImage()));
-        blogArticle.setTitleImage(null);
-        blogService.save(blogArticle);
-    }
-
-    private String saveImage(Project project, BlogArticle blogArticle, MultipartFile file, ImageType imageType) {
-        String imageFilePath = "";
-        try {
-            imageFilePath = fileSystemRepository.save(file.getBytes(), file.getOriginalFilename()); // NOSONAR
-
-            if (imageType != ImageType.BLOG_ARTICLE_TITLE_IMAGE)
-                saveImagePathToSpecifiedAttribute(imageType, project, imageFilePath);
-            else
-                saveImagePathForGivenBlogArticle(blogArticle, imageFilePath);
-
-            log.info("Image [" + file.getOriginalFilename() + "] was successfully uploaded.");
-        } catch (IOException ex) {
-            log.error(UPLOAD_FAILED_MESSAGE, ex);
-            throw new ImageUploadException(UPLOAD_FAILED_MESSAGE);
+    public void deleteImage(Image image) {
+        File file = ImageUtils.getImageFile(image);
+        if (!file.delete()) {
```

Comment:
tu mi Sonar vysvietil: `SonarLint: Use "java.nio.file.Files#delete" here for better messages on error conditions.`